control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -535,6 +587,111 @@ func (r *HostedControlPlaneReconciler) ensureControlPlane(ctx context.Context, h
 	return nil
 }
 
+func (r *HostedControlPlaneReconciler) reconcileEtcd(ctx context.Context, hcp *hyperv1.HostedControlPlane, releaseImage *releaseinfo.ReleaseImage) error {
+	rootCASecret := manifests.RootCASecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(rootCASecret), rootCASecret); err != nil {
+		return fmt.Errorf("cannot get root CA secret: %w", err)
+	}
+
+	// Etcd client secret
+	clientSecret := manifests.EtcdClientSecret(hcp.Namespace)
+	if err := r.Get(ctx, client.ObjectKeyFromObject(clientSecret), clientSecret); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
why do we need this Get request here? what is it giving us that is not already handled by `CreateOrUpdate`?